### PR TITLE
improve modelsearch

### PR DESCRIPTION
**Pull request by vrcAlbert**
- handel twinnable
- handle search on multiple fields
- allow custom config on autocomplete

### CODE REVIEW

**Comment by shaoshiva** [comment]
:+1:

**Comment by Foine** [comment]
:+1:

File: static/js/autocomplete.js [hunk 1]
```diff
@@ -162,7 +161,7 @@ define(['jquery-nos'], function ($nos) {
                                     // Get the results from an ajax query
                                     else {
                                         post.search = search;
-                                        $this.nosAjax({
+                                        $.ajax({
```
Comment (Foine) [inline static/js/autocomplete.js:164]: Why do you change this ?

**Comment by jguyomard** [comment]
Je trouve ça dommage de confondre title_property et search_property. Ce n'est pas parce qu'on veut chercher dans plusieurs champs qu'on veut forcément afficher tous ces champs concaténés comme label.

Un $search_property = $title_property si la méthode search_property() n'existe pas, puis un where sur  le/les champs de $search_property mais l'utilisation de $title_property en label ne serait pas plus adapté ?

**Comment by jguyomard** [comment]
(ouais, et je parle français si je veux #oups)

**Comment by vrcAlbert** [comment]
@jguyomard agree; but then how would I do if I want to search a Model_Person, which has two properties "first_name" and "last_name", and if I want them to appear has a full displayed name?
We often use "title_item()" method to perform this, but I can't use it in a query ^^

File: static/js/autocomplete.js (hunk 1 above)
Comment (vrcAlbert) [inline static/js/autocomplete.js:164]: nosAjax may add an "error" key that I don't want to be considered as a response in the autocomplete. 
As I don't call the Novius OS API with this ajax call, I don't need to use Novius OS tool to perform this.

**Comment by jguyomard** [comment]
Ok :/

**Pull request merged**